src/main/java/org/wildfly/security/evidence/BearerTokenEvidence.java

```diff
@@ -0,0 +1,21 @@
+package org.wildfly.security.evidence;
+
+import org.wildfly.common.Assert;
+
+/**
+ * An {@link Evidence} that usually holds a bearer security token.
+ *
+ * @author <a href="mailto:psilva@redhat.com">Pedro Igor</a>
+ */
+public class BearerTokenEvidence implements Evidence {
```

Comment:
I've removed that equals check from the realm to a instanceof. This ```BearerTokenEvidence``` should not be final, since we may have sub-types of it. For most cases, this interface is enough to get a token-based evidence working, but we may encounter situations where only the token is not enough and you may want to pass some more additional information.